taxtastic/subcommands/taxtable.py

```diff
@@ -30,20 +30,19 @@
 import sys
 from itertools import groupby
 
+from taxtastic.ncbi import RANK_LOOPS
 from taxtastic.taxonomy import Taxonomy
 from taxtastic.utils import add_database_args
 
 log = logging.getLogger(__name__)
 
 
-def replace_no_rank(ranks):
+def replace_loops(ranks):
     ranks = list(ranks)
-    while True:
-        try:
-            idx = ranks.index('no_rank')
+    for idx, r in enumerate(ranks[:]):
+        if r in RANK_LOOPS:
             ranks[idx] = ranks[idx - 1] + '_'
-        except ValueError:
-            return ranks
+    return ranks
 
```

Comment:
More micro-optimizations:

- ranks is being copied twice, once in list(ranks) and again in enumerate(ranks[:]) - do you need both?
- RANK_LOOPS is being accessed in the global scope, which is more expensive than as a local variable
- again, best to use a set to test membership

I'd suggest something like (untested):

```python
RANK_LOOPS_SET = set(RANK_LOOPS)

def replace_loops(ranks, replace=RANK_LOOPS_SET):
    for idx, r in enumerate(list(ranks)):
        if r in replace:
            ranks[idx] = ranks[idx - 1] + '_'
        return ranks
```

Reply:
* I was trying to change as little code as possible.  I am not sure about list(ranks) maybe an iterator was being passed at one point https://github.com/fhcrc/taxtastic/commit/89ee1ec44f8f0c4aa248b9d1d832011ce41cb7f9?

* I will update

* Thanks for the code, will update

Reply:
Updates:

1. Renamed RANK_LOOPS to UNORDERED_RANKS for clarity
1. Global references to RANK_LOOPS has been replaced with local variables except in taxonomy (see comment below)
1. self.unordered_ranks is a set()